ubireader/ubifs/nodes.py

```diff
@@ -183,13 +185,16 @@ class branch(object):
     Bin:buf     -- Raw data to extract header information from.
     """
     def __init__(self, buf):
-        fields = dict(list(zip(UBIFS_BRANCH_FIELDS, struct.unpack(UBIFS_BRANCH_FORMAT, buf))))
+        fields = dict(list(zip(UBIFS_BRANCH_FIELDS, struct.unpack(UBIFS_BRANCH_FORMAT, buf[0:UBIFS_BRANCH_SZ]))))
         for key in fields:
             if key == 'key':
                 setattr(self, key, parse_key(fields[key]))
             else:
                 setattr(self, key, fields[key])
 
+        if len(buf) > (UBIFS_BRANCH_SZ):
```

Comment:
Why is `UBIFS_BRANCH_SZ` between parentheses ?

Reply:
For no reason. I forgot to remove them. (Maybe in a previous iteration, this was `(UBIFS_BRANCH_SZ + something)` ?

I'll remove those unneeded parentheses.